pkg/logs/input/traps/tailer.go

```diff
@@ -0,0 +1,56 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://wwt.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package traps
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"github.com/DataDog/datadog-agent/pkg/snmp/traps"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Tailer consumes and processes a stream of trap packets, and sends them to a stream of log messages.
+type Tailer struct {
+	source     *config.LogSource
+	inputChan  traps.OutputChannel
+	outputChan chan *message.Message
+	done       chan bool
+}
+
+// NewTailer returns a new Tailer
+func NewTailer(source *config.LogSource, inputChan traps.OutputChannel, outputChan chan *message.Message) *Tailer {
+	return &Tailer{
+		source:     source,
+		inputChan:  inputChan,
+		outputChan: outputChan,
+		done:       make(chan bool, 1),
+	}
+}
+
+// Start starts the tailer.
+func (t *Tailer) Start() {
+	go t.run()
+}
+
+// Stop waits for the input buffer to be flushed.
+func (t *Tailer) Stop() {
+	<-t.done
+}
+
+func (t *Tailer) run() {
+	defer func() {
+		t.done <- true
+	}()
+
+	for packet := range t.inputChan {
+		content, err := FormatPacketJSON(packet)
+		if err != nil {
+			log.Errorf("failed to format packet: %s", err)
+			continue
+		}
+		t.outputChan <- message.NewMessageWithSource(content, message.StatusInfo, t.source)
```

Comment:
We may add some tagging logic here (switch to `message.NewMessage(content []byte, origin *Origin, status string)` and populate `origin` with tags like the sender address, the snmp version), it could be done later as it does not imply any refactoring.